backend/src/utils/proxy.ts

```diff
@@ -70,53 +71,71 @@ export const proxyService =
         const namespace = service.namespace ?? getParam(request, 'namespace');
         const name = getParam(request, 'name');
 
-        const doServiceRequest = () => {
+        const getServiceAddress = async () => {
           const scheme = tls ? 'https' : 'http';
+          const serviceName = `${service.prefix ?? ''}${name}${service.suffix ?? ''}`;
+          if (DEV_MODE) {
+            // Use port forwarding for local development:
+            // kubectl port-forward -n <namespace> svc/<service-name> <local.port>:<service.port>
+            return `${scheme}://${local.host}:${local.port}`;
+          }
+          const annotation = SERVICE_ADDRESS_ANNOTATIONS[service.addressType];
+          if (annotation) {
+            try {
+              const k8sService = (
+                await fastify.kube.coreV1Api.readNamespacedService(serviceName, namespace)
+              ).body;
+              const address = k8sService.metadata.annotations[annotation];
```

Comment:
annotations are optional.

How come typescript isn't complaining about this in our back end :(

Reply:
Oof, good question...

Reply:
I think our backend tsconfig doesn't have `strict` or `strictNullChecks`. Maybe we want a separate tech debt issue for that.